File: app/src/main/java/com/zuk0/gaijinsmash/riderz/utils/StationUtils.java
package com.zuk0.gaijinsmash.riderz.utils;

import com.zuk0.gaijinsmash.riderz.data.local.constants.StationList;

import java.util.Locale;
import java.util.Map;

public class StationUtils {

    public static String getStationNameFromAbbr(String abbr) {
        return StationList.stationMap.get(abbr.toLowerCase(Locale.US)); // keys are case sensitive
    }

    public static String getAbbrFromStationName(String name) {
        String result = "";
        for(Map.Entry<String, String> entry : StationList.stationMap.entrySet()){
            if(entry.getValue().equalsIgnoreCase(name)) {
               result = entry.getKey();
            }
        }
        return result;
    }
}

File: app/src/main/java/com/zuk0/gaijinsmash/riderz/utils/StationUtils.kt
package com.zuk0.gaijinsmash.riderz.utils

import com.zuk0.gaijinsmash.riderz.data.local.constants.StationList
import com.zuk0.gaijinsmash.riderz.utils.SharedPreferencesUtils
import java.util.*

object StationUtils {

    fun getStationNameFromAbbr(abbr: String): String? {
        return StationList.stationMap[abbr.toUpperCase(Locale.US)] // keys are case sensitive
    }

    fun getAbbrFromStationName(name: String?): String? {
        var result = ""
        for ((key, value) in StationList.stationMap) {
            if (value.equals(name, ignoreCase = true)) {
                result = key
            }
        }
        return result
    }

    fun validateStationName(name: String?): Boolean {
        return StationList.stationMap.containsValue(name)
    }
}